### PR TITLE
chore(observability)!: remove deprecated `component_name` metric tag

### PR DESCRIPTION
**Pull request by dsmith3197**
[OPW-94](https://datadoghq.atlassian.net/browse/OPW-94)

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
🎉

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ badges:
 Vector's 0.34.0 release includes **breaking changes**:
 
 1. [Removal of Deprecated Datadog Component Config Options](#datadog-deprecated-config-options)
+1. [Removal of Deprecated `component_name` Metric Tag](#deprecated-comoponent-name)
```
Comment (buraizu) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:15]:
```suggestion
1. [Removal of Deprecated `component_name` Metric Tag](#deprecated-component-name)
```

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md [hunk 2]
```diff
@@ -26,3 +27,7 @@ and Metrics sinks. Instead the `site` option should be used.
 
 The `region` and `application_key` config options have
 been removed from the Enterprise configuration. Instead of `region`, `site` should be used. `application_key` is no longer required.
+
+#### Removal of Deprecated `component_name` Metric Tag {#deprecated-comoponent-name}
```
Comment (buraizu) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:31]:
```suggestion
#### Removal of Deprecated `component_name` Metric Tag {#deprecated-component-name}
```

**Review by buraizu — APPROVED** [review]
I don't think this text is rendered, but approving with a minor suggested spelling update in case it helps with future doc maintenance

**Review by tobz — APPROVED** [review]

**Pull request merged**